### PR TITLE
[FLINK-33811] Fix the broken CI

### PR DESCRIPTION
**Pull request by Zakelly**
Fix FLINK-33811

Picked changes from 6.x and 8.x branchs of rocksdb. Since there are too many diffs between this repo and rocksdb, I only pick the useful lines from it without original commit messages.

## Change Log
 - Update the macos image version  (https://github.com/facebook/rocksdb/pull/11633)
 - Remove python env from macos build  (https://github.com/facebook/rocksdb/pull/9406)
 - Temporarily disable -Werror since we upgrade the image version, introducing lots of new warnings. (Only for frocksdb)
 - Remove .circleci/cat_ignore_eagain since we don't need it in new images (https://github.com/facebook/rocksdb/pull/9531)

### CODE REVIEW

**Comment by Zakelly** [comment]
@Myasuka @fredia PTAL, thanks a lot!

**Comment by Myasuka** [comment]
@Zakelly could you please provide the commits you refer to?

**Comment by Zakelly** [comment]
> @Zakelly could you please provide the commits you refer to?

I update them in description of this PR.

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,26 +34,28 @@ commands:
...
+            echo "export LZ4_DOWNLOAD_BASE=https://rocksdb-deps.s3.us-west-2.amazonaws.com/pkgs/lz4" >> $BASH_ENV
+            echo "export ZSTD_DOWNLOAD_BASE=https://rocksdb-deps.s3.us-west-2.amazonaws.com/pkgs/zstd" >> $BASH_ENV
 
   pre-steps-macos:
-      steps:
-        - pre-steps:
-            python-version: "3.7.8"
+    steps:
+      - pre-steps
+      # Temporarily disable warning as error since we upgrade the image version, and there are lots of new introduced warnings.
```
Comment (Myasuka) [inline .circleci/config.yml:57]: Could you please leave this change to a separate commit as we should try to fix the warnings in general cases.

File: .circleci/config.yml [hunk 2]
```diff
@@ -295,7 +289,7 @@ jobs:
       - pre-steps
       - install-gflags
       - install-clang-10
-      - run: COMPILE_WITH_UBSAN=1 OPT="-fsanitize-blacklist=.circleci/ubsan_suppression_list.txt" CC=clang-10 CXX=clang++-10 ROCKSDB_DISABLE_ALIGNED_NEW=1 USE_CLANG=1 make V=1 -j32 ubsan_check | .circleci/cat_ignore_eagain # aligned new doesn't work for reason we haven't figured out
```
Comment (Myasuka) [inline .circleci/config.yml:298]: Since `.circleci/cat_ignore_eagain` is useless, can we just delete that file like https://github.com/facebook/rocksdb/pull/9531?

**Review by Myasuka — COMMENTED** [review]
Thanks for the quick work, please take a look at my comments.

File: .circleci/config.yml (hunk 1 above)
Comment (Zakelly) [reply .circleci/config.yml:57]: Sure. But that's a big TODO since there are lots of warnings scattered everywhere. Those warnings are fixed by upstream rocksdb in 8.x version, so if we migrate to 8.9 or later, I suggest we leave the 6.20.3 branch as it is.

**Review by Myasuka — APPROVED** [review]
Thanks for the update, LGTM.

File: .circleci/config.yml (hunk 1 above)
Comment (Myasuka) [reply .circleci/config.yml:57]: Got it, agreed.

**Comment by fredia** [comment]
@Zakelly Thanks for your work, LGTM.
